virtool/tasks/data.py

```diff
@@ -0,0 +1,145 @@
+
+from sqlalchemy import select
+from sqlalchemy.ext.asyncio import AsyncEngine, AsyncSession
+from virtool_core.models.task import Task as TaskModel
+
+import virtool.utils
+from virtool.data.errors import ResourceNotFoundError
+from virtool.tasks.models import Task
+
+
+class TasksData:
+    def __init__(self,  pg: AsyncEngine):
+        self._pg = pg
+
+    async def find(self) -> list:
+        """
+        Get a list of all tasks.
+
+        :return: a list of task records
+
+        """
+        documents = []
+        async with AsyncSession(self._pg) as session:
+            tasks = (await session.execute(select(Task))).scalars().all()
+            for task in tasks:
+                documents.append(TaskModel(**task.to_dict()))
+        return documents
+
+    async def get(self, task_id: int) -> TaskModel:
+        """
+        Get the task corresponding with passed "task_id".
+
+        :param task_id: ths ID of the task
+        :return: a task record
+
+        """
+        async with AsyncSession(self._pg) as session:
+            result = (await session.execute(select(Task).filter_by(id=task_id))).scalar()
+
+        if result is None:
+            raise ResourceNotFoundError
+
+        return TaskModel(**result.to_dict())
+
+    async def register(self, task_class, context: dict = None) -> TaskModel:
+        """
+        Create a new task record and store it.
+
+        :param task_class: a subclass of a Virtool :class:`~virtool.tasks.task.Task`
+        :param context: A dict containing data used by the task
+
+        :return: the new task record
+
+        """
+        task = Task(
+            complete=False,
+            context=context or {},
+            step=task_class.task_type,
+            count=0,
+            created_at=virtool.utils.timestamp(),
+            progress=0,
+            type=task_class.task_type,
+        )
+        async with AsyncSession(self._pg) as session:
+            session.add(task)
+            await session.flush()
+            document = task.to_dict()
+            await session.commit()
+        return TaskModel(**document)
```

Comment:
Use blank lines to break up ideas.

```suggestion
        task = Task(
            complete=False,
            context=context or {},
            step=task_class.task_type,
            count=0,
            created_at=virtool.utils.timestamp(),
            progress=0,
            type=task_class.task_type,
        )
        
        async with AsyncSession(self._pg) as session:
            session.add(task)
            await session.flush()
            document = task.to_dict()
            await session.commit()
            
        return TaskModel(**document)
```